pkg/machine/applehv/stubber.go

```diff
@@ -0,0 +1,331 @@
+	if err != nil {
+		return nil, nil, err
+	}
+
+	vm.Devices = append(vm.Devices, defaultDevices...)
+	vm.Devices = append(vm.Devices, netDevice)
+
+	mounts, err := virtIOFsToVFKitVirtIODevice(mc.Mounts)
+	if err != nil {
+		return nil, nil, err
+	}
+	vm.Devices = append(vm.Devices, mounts...)
+
+	// To start the VM, we need to call vfkit
+	cfg, err := config.Default()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	vfkitBinaryPath, err := cfg.FindHelperBinary(vfkitCommand, true)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	logrus.Debugf("vfkit path is: %s", vfkitBinaryPath)
+
+	cmd, err := vm.Cmd(vfkitBinaryPath)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	vfkitEndpointArgs, err := getVfKitEndpointCMDArgs(mc.AppleHypervisor.Vfkit.Endpoint)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	machineDataDir, err := mc.DataDir()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	cmd.Args = append(cmd.Args, vfkitEndpointArgs...)
+
+	// vfkit
+	//--cpus 4
+	//--memory 2048
+	//--bootloader efi,variable-store=/Users/brentbaude/.local/share/containers/podman/machine/applehv/podman-machine-defaults,create
+	//--device virtio-blk,path=/Users/brentbaude/.local/share/containers/podman/machine/applehv/podman-machine-default_fedora-coreos-39.20240112.2.0-applehv.aarch64.raw
+	//--device virtio-rng
+	//--device virtio-serial,logFilePath=/Users/brentbaude/.local/share/containers/podman/machine/applehv/podman-machine-default.log
+	//--device virtio-vsock,port=1025,socketURL=/var/folders/qp/hh156k710d96xc26yd0947v00000gn/T/podman/podman-machine-default_ready.sock,listen
+	//--device virtio-net,unixSocketPath=/var/folders/qp/hh156k710d96xc26yd0947v00000gn/T/podman/gvproxy.sock,mac=5a:94:ef:e4:0c:ee
+	//--device virtio-fs,sharedDir=/Users,mountTag=Users
+	//--device virtio-fs,sharedDir=/private,mountTag=private
+	//--device virtio-fs,sharedDir=/var/folders,mountTag=var-folders
+	//--restful-uri tcp://localhost:50212
+	//--device virtio-vsock,port=1024,socketURL=/Users/brentbaude/.local/share/containers/podman/machine/applehv/ignition.sock,listen
+	//--device virtio-gpu,width=800,height=600
+	//--device virtio-input,pointing
+	//--device virtio-input,keyboard
+	//--gui]
+
+	firstBoot, err := mc.IsFirstBoot()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if firstBoot {
+		// If this is the first boot of the vm, we need to add the vsock
+		// device to vfkit so we can inject the ignition file
+		socketName := fmt.Sprintf("%s-%s", mc.Name, ignitionSocketName)
+		ignitionSocket, err = machineDataDir.AppendToNewVMFile(socketName, &socketName)
+		if err != nil {
+			return nil, nil, err
+		}
+		if err := ignitionSocket.Delete(); err != nil {
+			logrus.Errorf("unable to delete ignition socket: %q", err)
+		}
+
+		ignitionVsockDeviceCLI, err := getIgnitionVsockDeviceAsCLI(ignitionSocket.GetPath())
+		if err != nil {
+			return nil, nil, err
+		}
+		cmd.Args = append(cmd.Args, ignitionVsockDeviceCLI...)
+	}
+
+	if logrus.IsLevelEnabled(logrus.DebugLevel) {
```

Comment:
can we move this above the first `if firstBoot {...` check and then add the following `firstBoot` check's code to just one if-statement?